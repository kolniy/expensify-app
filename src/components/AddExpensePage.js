import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { addExpense } from '../actions/expense'

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
     this.props.addExpense(expense)
     this.props.history.push('/')
   }

   render(){
      return (
         <div>
       <h2>Add Expense</h2>
       <ExpenseForm onSubmit={this.onSubmit} />
    </div>
      )
   }
}

const matchDispatchToProps = (dispatch) => (
   {
      addExpense: (expense) => dispatch(addExpense(expense))
   }
)

export default connect(undefined, matchDispatchToProps)(AddExpensePage)