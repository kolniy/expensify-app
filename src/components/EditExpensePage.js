import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, startRemoveExpense } from '../actions/expense'

export class EditExpensePage extends React.Component {
    editExpense = (expense) => {
        this.props.editExpense(this.props.expense.id, expense)
        this.props.history.push('/')
    }
    startRemoveExpense = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/')
    }
    render(){
        if(this.props.expense === undefined){
            this.props.history.push('/')
        }
        return (
            <div>
            <ExpenseForm 
             expense={this.props.expense} 
             onSubmit={(expense) => {
              this.editExpense(expense)
            }}
            />
            <button onClick={this.startRemoveExpense}>Remove</button>
        </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id ),
    expenses: state.expenses
})

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    startRemoveExpense: ({ id }) => dispatch(startRemoveExpense({ id }))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)