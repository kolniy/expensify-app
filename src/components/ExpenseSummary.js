import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import expenseSelector from '../selectors/expenses'
import expenseTotal from '../selectors/expense-total'

export const ExpenseSummary = (prop) => {
    return (
        <div>
            {
            prop.expenseCount === 1 ? (
            <p>Viewing {prop.expenseCount} expense totalling {numeral(prop.expenseTotal / 100).format('$0,0.00')}</p>) :
             (<p>Viewing {prop.expenseCount} expenses totalling {numeral(prop.expenseTotal / 100).format('$0,0.00')}
             </p>)
             }
        </div>
    )
}

const mapStateToProp = (state) => {
    return {
        expenseCount: expenseSelector(state.expenses, state.filters).length,
        expenseTotal: expenseTotal(expenseSelector(state.expenses, state.filters))
    }
}

export default connect(mapStateToProp)(ExpenseSummary)