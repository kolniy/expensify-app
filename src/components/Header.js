import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogOut } from '../actions/auth'

export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard </NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense </NavLink>
        <button onClick={startLogOut}>Log Out</button>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogOut : () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatchToProps)(Header)