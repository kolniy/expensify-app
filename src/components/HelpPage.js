import React from 'react'

const HelpPage = () => (
    <div>
        this is my Help Expense component page
    </div>
)

export default HelpPage