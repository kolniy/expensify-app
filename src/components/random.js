function viralAdvertising(n) {
    let numOfPeopleOnFirstDay = 5
    let sum = 0

    for(var i = 1; i <= n; i++){
        let numOfPeopleOnsubsequentDays =  numOfPeopleOnFirstDay
        let numOfLikeOnEachDay = Math.floor(numOfPeopleOnsubsequentDays/2)
        sum += numOfLikeOnEachDay
        numOfPeopleOnFirstDay = numOfLikeOnEachDay * 3
    }

    return sum
}

console.log(viralAdvertising(20))