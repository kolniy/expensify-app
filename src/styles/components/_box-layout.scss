.box-layout {
    background: url('/images/bg.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;

    &__box {
      background: rgba($color: #fff, $alpha: .7);
      border-radius: 3px;
      padding: $l-size $m-size;
      text-align: center;
      width: 25rem;
      box-shadow: 0 3rem 6rem rgba($color: #fff, $alpha: .14);
    }

    &__title {
        margin : 0 0 $m-size 0;
        line-height: 1;
        color: #333;
    }
}
