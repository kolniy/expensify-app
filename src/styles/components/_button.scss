.button {
    background-color: $blue;
    border-radius: 3px;
    border: none;
    color: #fff;
    display: inline-block;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;

    &--link {
        background: none;
    }

    &--secondary{
        background-color: #888888;
    }
}