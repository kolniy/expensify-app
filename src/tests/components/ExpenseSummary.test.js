import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should correctly render summary for one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={230} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render summary for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={2300} />)
    expect(wrapper).toMatchSnapshot()
})
