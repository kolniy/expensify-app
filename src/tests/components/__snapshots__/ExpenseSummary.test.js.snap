// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $23.00
  </p>
</div>
`;

exports[`should correctly render summary for one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense totalling 
    $2.30
  </p>
</div>
`;
